script/deploy/DeployChainlinkRelayers.sol

```diff
@@ -0,0 +1,101 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.18;
+
+import { console } from "forge-std-next/console.sol";
+import { Script } from "script/utils/mento/Script.sol";
+import { Chain } from "script/utils/mento/Chain.sol";
+import { Contracts } from "script/utils/mento/Contracts.sol";
+
+import { ChainlinkRelayerFactory } from "mento-core-develop/oracles/ChainlinkRelayerFactory.sol";
+import { ChainlinkRelayerFactoryProxy } from "mento-core-develop/oracles/ChainlinkRelayerFactoryProxy.sol";
+import { ChainlinkRelayerFactoryProxyAdmin } from "mento-core-develop/oracles/ChainlinkRelayerFactoryProxyAdmin.sol";
+import { IChainlinkRelayer } from "mento-core-develop/interfaces/IChainlinkRelayer.sol";
+
+contract DeployChainlinkRelayers is Script {
+  using Contracts for Contracts.Cache;
+
+  struct Relayer {
+    string rateFeed;
+    IChainlinkRelayer.Config config;
+  }
+
+  Relayer[] relayers = [
+    Relayer({
+      rateFeed: "chainlink:CELO/USD:v1",
+      config: singleAggConfig(contracts.dependency("Chainlink.CELOUSD"), false)
+    }),
+    Relayer({
+      rateFeed: "chainlink:USDT/USD:v1",
+      config: singleAggConfig(contracts.dependency("Chainlink.USDTUSD"), false)
+    }),
+    Relayer({
+      rateFeed: "chainlink:CELO/PHP",
+      config: IChainlinkRelayer.Config({
+        maxTimestampSpread: 1000,
+        chainlinkAggregator0: contracts.dependency("Chainlink.CELOUSD"),
+        chainlinkAggregator1: contracts.dependency("Chainlink.PHPUSD"),
+        chainlinkAggregator2: address(0),
+        chainlinkAggregator3: address(0),
+        invertAggregator0: false,
+        invertAggregator1: true,
+        invertAggregator2: false,
+        invertAggregator3: false
+      })
+    }),
+    Relayer({ rateFeed: "PHP/USD", config: singleAggConfig(contracts.dependency("Chainlink.PHPUSD"), false) })
+  ];
+
+  ChainlinkRelayerFactory relayerFactory;
+
+  constructor() Script() {
+    contracts.load("ChainlinkRelayerFactory", "checkpoint");
+    relayerFactory = ChainlinkRelayerFactory(contracts.deployed("ChainlinkRelayerFactoryProxy"));
+  }
+
+  function run() public {
+    vm.startBroadcast(Chain.deployerPrivateKey());
+    {
+      for (uint i = 0; i < relayers.length; i++) {
+        deployRelayerIfNoneOrDifferent(toRateFeedId(relayers[i].rateFeed), relayers[i].config);
+      }
+    }
+    vm.stopBroadcast();
+
+    for (uint i = 0; i < relayers.length; i++) {
+      address rateFeedId = toRateFeedId(relayers[i].rateFeed);
+      address relayer = relayerFactory.getRelayer(rateFeedId);
+      console.log(relayers[i].rateFeed, rateFeedId, relayer);
+    }
+  }
+
+  function deployRelayerIfNoneOrDifferent(address rateFeedId, IChainlinkRelayer.Config memory config) internal {
```

Comment:
tbh i didn't fully understand the `IfNoneOrDifferent` suffix here, could you elaborate?

Reply:
ok, think i got it now, only deploy if no current relayer exists in factory storage OR if a relayer exists but with a different address. off the top of my head don't have a better naming suggestion but feels like there could be one :))